EpsilonAuxBmsV2/Tests/tests/OrionInterfaceTest/OrionInterfaceTest.c

```diff
@@ -17,241 +12,75 @@ osThreadId_t commonContactorGatekeeperTaskHandle;
-    expectedAuxStatus.allowDischarge = 0;
-    expectedAuxStatus.strobeBmsLight = 1;
-    expectedAuxStatus.dischargeShouldTrip = 1;
-    expectedAuxStatus.chargeShouldTrip = 1;
-
+// void test_onlyUpdateAllowCharge() {
+//     AuxStatus auxStatusToUpdate;
+//     auxStatusToUpdate.allowCharge = 1;
+//     message.highCellVoltage = 42000;
+//     updateAllowChargeAndAllowDischarge(&message, &auxStatusToUpdate);
+//     TEST_ASSERT_EQUAL_MESSAGE(0, auxStatusToUpdate.allowCharge, "auxStatusToUpdate.allowCharge failed to update");
+// }
+
+// void test_onlyUpdateAllowDischarge() {
+//     AuxStatus auxStatusToUpdate;
+//     auxStatusToUpdate.allowDischarge = 1;
+//     message.lowCellVoltage = 25000;
+//     updateAllowChargeAndAllowDischarge(&message, &auxStatusToUpdate);
+//     TEST_ASSERT_EQUAL_MESSAGE(0, auxStatusToUpdate.allowDischarge, "auxStatusToUpdate.allowDischarge failed to update");
+// }
+
+// TODO: Find a better naming convetion for unit tests
+void test_osMessageQueueGetTimeout() {
+    AuxStatus expectedAuxStatus = {.orionCanReceivedRecently = 0};
+    auxBmsContactorState.startupDone = 0;
+    uint32_t contactorControlEventFlags = 0;
+    
+    osMessageQueueGet_ExpectAndReturn(orionInterfaceQueue, &message, NULL, ORION_QUEUE_TIMEOUT, osErrorTimeout);
+    HAL_GPIO_ReadPin_ExpectAndReturn(ORION_DISCHARGE_ENABLE_SENSE_GPIO_Port, ORION_DISCHARGE_ENABLE_SENSE_Pin, 1);
+    HAL_GPIO_ReadPin_ExpectAndReturn(ORION_CHARGE_ENABLE_SENSE_GPIO_Port, ORION_CHARGE_ENABLE_SENSE_Pin, 1);
+    osMutexAcquire_ExpectAndReturn(auxTripMutex, MUTEX_TIMEOUT, osError);
+    osMutexAcquire_ExpectAndReturn(auxStatusOrionInterfaceMutex, MUTEX_TIMEOUT, osOK);
+    osMutexRelease_ExpectAndReturn(auxStatusOrionInterfaceMutex, 0);
+    osEventFlagsSet_ExpectAndReturn(contactorControlEventBits, contactorControlEventFlags, 0);
+    
     orionInterface(&message);
-
-    assertAuxStatusEqual(expectedAuxStatus);
+    TEST_ASSERT_EQUAL_MESSAGE(expectedAuxStatus.orionCanReceivedRecently, auxStatus.orionCanReceivedRecently, "auxStatus.orionCanReceivedRecently failed to update");
 }
 
-// Orion reports a cell voltage above the internal Aux BMS range
-// Test that the charge contactor off event is triggered & statuses are correct
-void test_TooHighMaxCellVoltageShouldTurnOffChargeAndUpdateStatuses()
-{
-    setOrionInterfaceOsExpects(1);
-    setTripExpects(0, 0, 0);
-    setContactorState(CLOSED, CLOSED, 1);
-    setContactorEventFlagsExpects(CHARGE_OFF);
+void test_shouldDisconnectContactors() {
+    AuxStatus expectedAuxStatus = {
+        .orionCanReceivedRecently = 0,
+        .allowCharge = 0,
+        .allowDischarge = 0,
+        .strobeBmsLight = 1
+    };
+    AuxBmsContactorState expectedAuxBmsContactorState = {.contactorsDisconnected = 1};
+    uint32_t contactorControlEventFlags = 0x28; // CHARGE_OPENED and DISCHARGE_OPENED
+
+    osMessageQueueGet_ExpectAndReturn(orionInterfaceQueue, &message, NULL, ORION_QUEUE_TIMEOUT, osErrorTimeout);
```

Comment:
Not sure if you meant for this to return `osErrorTimeout`

Reply:
```
if (status == osErrorTimeout) // If waiting on the queue times out
{
    //Set orion can message recieved recently to 0
    localAuxStatus.orionCanReceivedRecently = 0;
}
else
{
    // Determine trip conditions and contactor settings based on Orion CAN
    localAuxStatus.orionCanReceivedRecently = 1;
    updateAllowChargeAndAllowDischarge(message, &localAuxStatus);
    updateAuxTrip(message, &localAuxTrip);
    localAuxStatus.dischargeShouldTrip = checkDischargeTrip(localAuxTrip);
    localAuxStatus.chargeShouldTrip = checkChargeTrip(localAuxTrip);
    shouldDisconnectContactors = localAuxTrip.protectionTrip
                                || localAuxStatus.dischargeShouldTrip
                                || localAuxStatus.chargeShouldTrip;
}
```

We were not able to test this else block because of the localAuxTrip being a local variable. That's why we were returning `osErrorTimeout`. Not sure if there is a way to test this particular block, but we did test most of these functions in TripTest.c.

Reply:
I can think of two ways:
- Don't mock the Trip functions. This will make testing incredibly messy
- Make `localAuxTrip` not local to the function. This is less clean in the actual design code, but cleaner for testing

Either way, I think it's very critical to test this point as this has all the trip functionality.
@bill-luu what do you think?